### PR TITLE
Oakland predicates

### PR DESCRIPTION
**Pull request by dionyziz**
Depends #250

### CODE REVIEW

File: etc/theory/compression_security.tex [hunk 1]
```diff
@@ -571,16 +571,18 @@ \subsection{Length monotonicity}\label{subsec:lenmonotone}
 \section{Compressibility with respect to predicates}\label{sec:propertycom}
 
 Let $\mathcal{M}$ be a plaintext distribution and $Q(s)$ be a predicate on the
-plaintext $s \in \mathcal{M}$. Let us use $\mathcal{M}_Q$ to denote the subset
-of $\mathcal{M}$ restricted over the elements for which $Q$ holds
-proportionally retaining measure and assuming $\mathcal{M}_Q \neq \emptyset$
-and $\mathcal{M}_{\lnot Q} \neq \emptyset$.
+plaintext $s$ drawn from $\mathcal{M}$. Furthermore, require that $Q$ is not a
```
Comment (dimkarakostas) [inline etc/theory/compression_security.tex]: _we_ require

File: etc/theory/compression_security.tex [hunk 2]
```diff
@@ -641,8 +641,9 @@ \section{Compressibility with respect to predicates}\label{sec:propertycom}
 Com, \mathcal{M}, \mathcal{V})$ holds.
 
 The compression-detectability property exists in all common compression
-functions as long as $s$ and $r$ compress in the same context. We will
-illustrate such examples later in this work.
+functions as long as $s$ and $r$ compress in the same context. We will now
+prove the fact that all compression functions which are ideal under some
+distribution exhibit compression-detectability of some predicate $Q$.
```
Comment (dimkarakostas) [inline etc/theory/compression_security.tex]: some distribution \mathcal{M} exhibit

File: etc/theory/compression_security.tex [hunk 3]
```diff
@@ -582,39 +615,35 @@ \section{Compressibility with respect to predicates}\label{sec:propertycom}
...
 
-Furthermore we require that such a pair $\overbar{r}$ is polynomially
-computable:
-\begin{align*}
-    \exists \text{PPT } \mathcal{O}_R:\\
-    dct(Q, \overbar{r}, f, Com, \mathcal{M}, \mathcal{V}),\\
-    \overbar{r} \leftarrow \mathcal{O}_R(Com, f, Q, \mathcal{M}, \mathcal{V})
-\end{align*}
+We call $Q$ \textit{compression-detectable} with an advantage $\alpha$ if a
+pair $\overbar{r} = (r_1, r_2)$ exists, such that $dct_\alpha(Q, \overbar{r},
```
Comment (dimkarakostas) [inline etc/theory/compression_security.tex]: :s/dct/dtc

**Comment by dionyziz** [comment]
LGTY for both #251 and #250?

**Comment by dimkarakostas** [comment]
Yes, fix the comments I added and merge.

**Pull request merged**